src/daemon/rpmostree-sysroot-upgrader.c

```diff
@@ -1069,9 +1070,30 @@ perform_local_assembly (RpmOstreeSysrootUpgrader *self,
       kernel_state = rpmostree_find_kernel (self->tmprootfs_dfd, cancellable, error);
       if (!kernel_state)
         return FALSE;
+
+      /* note we extract the initramfs path here but we only use it as a fallback to use
+       * with `--rebuild` if we're missing `initramfs-args` in the commit metadata -- in the
+       * future, we could remove this entirely */
       g_variant_get (kernel_state, "(&s&s&sm&s)",
                      &kver, &bootdir,
                      &kernel_path, &initramfs_path);
+
+      /* also try to extract initramfs args from the base commit */
+      g_autoptr(GVariant) base_commit = NULL;
+      if (!ostree_repo_load_variant (self->repo, OSTREE_OBJECT_TYPE_COMMIT,
+                                     self->base_revision, &base_commit, error))
```

Comment:
Hmm.  So `self->base_revision` is only set if we're already layered.  I think this code might only work on *subsequent* upgrades?

Can you add at least a `g_assert (self->base_revision);` here even if I'm wrong just as a sanity check?

Reply:
Sure, added. But yes, we always set `self->base_revision`. In the non-layered case, it just matches the target deployment checksum.

Reply:
OK right, I see that in the code now.